File: lily-web/src/types/action.ts
export type ActionStatus = 'PENDING' | 'APPROVED' | 'RUNNING' | 'SUCCESS' | 'FAILED' | 'REJECTED'

export type ActionType = 
  | 'RESTART_SERVICE'
  | 'CLEAR_CACHE'
  | 'SCALE_SERVICE'
  | 'ROLLBACK_DEPLOY'
  | 'KILL_IDLE_CONNECTIONS'
  | 'FLUSH_QUEUE'
  | 'BLOCK_IP'
  | 'OTHER'

export interface IncidentAction {
  id: string
  incidentId: string
  incident_id?: string
  incidentTitle?: string
  actionType: ActionType | string
  action_type?: string
  type?: string
  description: string
  status: ActionStatus
  triggeredBy: string
  triggered_by?: string
  requestedBy?: string | null
  requested_by?: string | null
  approvedBy?: string | null
  approved_by?: string | null
  startedAt?: string | null
  started_at?: string | null
  completedAt?: string | null
  completed_at?: string | null
  result?: string | null
  errorMessage?: string | null
  error_message?: string | null
  createdAt: string
  created_at?: string
}

File: lily-backend/app/schemas/action.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.utils.enums import ActionStatus, ActionType


class ActionCreate(BaseModel):
    action_type: ActionType
    description: str = Field(min_length=1)
    requested_by: str | None = None


class ActionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    incident_id: uuid.UUID
    action_type: ActionType
    description: str
    status: ActionStatus
    requested_by: str | None
    approved_by: str | None
    started_at: datetime | None
    completed_at: datetime | None
    result: str | None
    error_message: str | None